dev/NavigationView/NavigationView.cpp

```diff
@@ -215,45 +265,43 @@ void NavigationView::OnApplyTemplate()
 
     m_topNavGrid.set(GetTemplateChildT<winrt::Grid>(c_topNavGrid, controlProtected));
 
+    UpdateNavigationViewItemsFactory();
+
     // Change code to NOT do this if we're in top nav mode, to prevent it from being realized:
-    if (auto leftNavListView = GetTemplateChildT<winrt::ListView>(c_menuItemsHost, controlProtected))
+    if (auto leftNavRepeater = GetTemplateChildT<winrt::ItemsRepeater>(c_menuItemsHost, controlProtected))
     {
-        m_leftNavListView.set(leftNavListView);
-
-        m_leftNavListViewLoadedRevoker = leftNavListView.Loaded(winrt::auto_revoke, { this, &NavigationView::OnListViewLoaded });
+        m_leftNavRepeater.set(leftNavRepeater);
 
-        m_leftNavListViewSelectionChangedRevoker = leftNavListView.SelectionChanged(winrt::auto_revoke, { this, &NavigationView::OnSelectionChanged });
-        m_leftNavListViewItemClickRevoker = leftNavListView.ItemClick(winrt::auto_revoke, { this, &NavigationView::OnItemClick });
-
-        SetNavigationViewListPosition(leftNavListView, NavigationViewListPosition::LeftNav);
+        m_leftNavItemsRepeaterElementPreparedRevoker = leftNavRepeater.ElementPrepared(winrt::auto_revoke, { this, &NavigationView::RepeaterElementPrepared });
+        m_leftNavItemsRepeaterElementClearingRevoker = leftNavRepeater.ElementClearing(winrt::auto_revoke, { this, &NavigationView::RepeaterElementClearing });
+ 
+        m_leftNavRepeaterLoadedRevoker = leftNavRepeater.Loaded(winrt::auto_revoke, { this, &NavigationView::OnRepeaterLoaded });
 
-        // Since RS5, SingleSelectionFollowsFocus is set by XAML other than by code
-        if (SharedHelpers::IsRS1OrHigher() && ShouldPreserveNavigationViewRS4Behavior())
-        {
-            leftNavListView.SingleSelectionFollowsFocus(false);
-        }
+        leftNavRepeater.ItemTemplate(*m_navigationViewItemsFactory);
     }
 
     // Change code to NOT do this if we're in left nav mode, to prevent it from being realized:
```

Comment:
Does this comment and the one above still make sense to keep ?

Reply:
It is still applicable, potentially a change we can make in the future